src/font.c

```diff
@@ -1152,14 +1160,12 @@ GdipPrivateAddMemoryFont(GpFontCollection *fontCollection, GDIPCONST void *memor
 	}
 	close(f);
 
-	/* FIXME - this doesn't seems to catch "bad" (e.g. invalid) font files */
-	result = FcConfigAppFontAddFile (fontCollection->config, fontfile);
```

Comment:
It appears that GDI+ just returns Ok for invalid files (added a test for that). Previously, libgdiplus would seemingly non-deterministically return `FileNotFound`.

In both libraries the count stays at the previous number if the font file is invalid.